### DISCUSSION TITLE
how to add login/password to go2rtc streams

### DISCUSSION THREAD

**Discussion by teredactle** [gtxaspec/wz_mini_hacks#708] [Q&A]
Hi there, is there an easy way to add a "global" loing/password for the go2rtc streams that are available by default?

by default if you click on Streams (once you go to http://IP:1984) there are 1080p/360p links to streams, and when you go there non have a login/password.

There is a config button that shows the code, I'm not getting how I can put a global login/password via that editor.
Is it even possible?

Thanks

**Comment by gtxaspec** [comment]
did you check the go2rtc manual?  our wiki artcle https://github.com/gtxaspec/wz_mini_hacks/wiki/Video-Streaming#go2rtc-information has a link to the go2rtc repo which describes how to add password authentication.

**Comment by rocket59** [comment]
Posting the config link here as both a response and reminder to myself when I forget ! 
https://github.com/AlexxIT/go2rtc/wiki/Configuration

The yaml file can be found at SD_ROOT/wz_mini/etc/. You can also edit the file via the go2rtc web gui.

BTW - thanks for all your efforts on this @gtxaspec and to AlexxIT! The new go2rtc server got rid of all the artifacts from my V3 camera in Frigate.

**Comment by teredactle** [comment]
Thanks for posting that, I changed via the webui, so for RTSP u/p it worked great.
However, for the other streams via (html) link, I understand these are Webrtc links to the stream so I used the following code but does not prompt me for login/password.

webrtc:
  listen: ":8555"
      username: ""
      credential: ""

How do I secure the HTML links?

![image](https://github.com/gtxaspec/wz_mini_hacks/assets/37247773/a1cffcb4-6aea-49ee-982d-018314a33045)

**Comment by gtxaspec** [comment]
see https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#module-api

**Reply by Fmstrat** [reply]
> see https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#module-api

I'm confused, this doesn't seem to answer the question? The `api` flag in the YML is for securing the web portal, not the WebRTC port. There doesn't seem to be a solution currently for WebRTC credentials? https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#module-webrtc

**Comment by Fmstrat** [comment]
Ok, I've figured this out. The `api` tag in `go2rtc.yml` impacts all aspects, including the WebRTC stream. For anyone who wants to enable user/pass on the WebRTC streams, SSH in and edit `/media/mmc/wz_mini/etc/go2rtc.yml` (this will also edit the file in `/opt/wz_mini/etc` as they are simlinks, and add this to the root (far left, no indention) of the file:

```yml
api:
  username: "admin"
  password: "mypass"
```

Then restart go2rtc:

```bash
/opt/wz_mini/etc/network.d/S17go2rtc stop
/opt/wz_mini/etc/network.d/S17go2rtc start
```

You can then access the camera and are forced to type a user/pass, or you can use the URL format:

```
http://user:pass@host:port/webrtc.html?src=360p&media=video+audio
```
